nomad/structs/services.go

```diff
@@ -1422,6 +1426,10 @@ func (t *SidecarTask) Equal(o *SidecarTask) bool {
 		return false
 	}
 
+	if !slices.Equal(t.VolumeMounts, o.VolumeMounts) {
```

Comment:
slices.Equal takes into account the order of the elements, should we order them before comparing?

Reply:
Actually, I had a look for how we treat this elsewhere and we always use `!slices.Equal`, even if the ordering isn't important. That lets operators force a job update without changing anything important, so I suspect we want to keep that behavior intact.